lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -60,7 +58,13 @@ def test_query(
         headers = None
 
         if user:
-            token = self.token(client, server_config, user)
+            if user == "test_admin":
+                token = pbench_admin_token
+            else:
+                response = login_user(client, server_config, user, "12345")
+                assert response.status_code == HTTPStatus.OK
+                token = response.json["auth_token"]
+            assert token
```

Comment:
I prefer to avoid levels of indentation when plausible.  (There are several reasons for this; among them are the challenges of fitting code on the screen for the reader, but the structure here also affects the unit test code.)  

So,
```
        if user == "test_admin":
            token = pbench_admin_token
        elif user:
            response = login_user(client, server_config, user, "12345")
            assert response.status_code == HTTPStatus.OK
            token = response.json["auth_token"]
            assert token
```
That is, if `user` is `None`, then we can still compare it to the literal string, and we won't enter the first `if` block;  then we'll do the same test as you had in the original code and be done.  If the user is `"test_admin"` we get through with one fewer conditional tests; and if the `user` is something else, then we do the same number of tests (just in the opposite order).

(We do lose the ability to confirm that the `pbench_admin_token` fixture returned a value, but that's probably more of a gain than a loss....)